### PR TITLE
[RNMobile] Introduce a common API to style Button from both web and mobile

### PR DESCRIPTION
**Pull request by dratwas**
## Description
This is an exploration of x-platform style system. https://github.com/wordpress-mobile/gutenberg-mobile/issues/1411
This is the second iteration on https://github.com/wordpress-mobile/gutenberg-mobile/issues/1386

Gutenberg-mobile PR - https://github.com/wordpress-mobile/gutenberg-mobile/pull/1684

In this PR I used `styled-system` with `emotion` to create a primitive Button component that could be used in web and mobile and could be styled via props. That PR contains the mobile part.

- I added the `button` primitive component (`TouchableOpacity`) that can be used with `styled-system`
- I added the theme provider and theme values (they can be a bit outdated since I took them from my previous PR https://github.com/WordPress/gutenberg/pull/17614 )
- I used created primitive in `@components/Button`
- I added `isSmall`, `isLarge` and `isPrimary` props and set button style related to them (just to demonstrate)
- I used Button primitive in breadcrumbs
- I set `isLarge` and `isPrimary` to `@components/Button` in `ButtonBlockAppender` component to demonstrate they work :)

## How has this been tested?
- Run `gutenberg-mobile` app
- All buttons should look the same as before
- Check if breadcrumbs are rendered properly
- Check if block appender has `blue` background and additional padding (because of `isLarge` and `isPrimary`)

### without `isLarge` and `isPrimary`
Simulator Screen Shot - iPhone X - 2019-12-16 at 12 06 35

### with `isLarge` and `isPrimary`
Simulator Screen Shot - iPhone X - 2019-12-16 at 13 52 30

## Types of changes
Experimenting with `emotion` and `styled-system`

### CODE REVIEW

File: packages/block-editor/src/components/block-list/breadcrumb.native.js [hunk 1]
```diff
@@ -27,8 +28,9 @@ const BlockBreadcrumb = ( {
 } ) => {
 	return (
 		<View style={ styles.breadcrumbContainer }>
-			<TouchableOpacity
-				style={ styles.button }
+			<PrimitiveButton
```
Comment (dratwas) [inline packages/block-editor/src/components/block-list/breadcrumb.native.js]: These changes are only for demonstrating how to style component using`styled-system` props

File: packages/block-editor/src/components/button-block-appender/index.native.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ function ButtonBlockAppender( {
 				rootClientId={ rootClientId }
 				renderToggle={ ( { onToggle, disabled, isOpen } ) => (
 					<Button
+						isLarge
```
Comment (dratwas) [inline packages/block-editor/src/components/button-block-appender/index.native.js]: For demonstrating that `isPrimary` and `isLarge` works

**Comment by pinarol** [comment]
>I added isSmall, isLarge and isPrimary props and set button style related to them (just to demonstrate)
>I used Button primitive in breadcrumbs
>I set isLarge and isPrimary to @components/Button in ButtonBlockAppender component to demonstrate they work :)

Just to confirm, this PR doesn't include real implementations of these props. But just demonstrates a way about how to use such props to customize style.

**Comment by gziolo** [comment]
> Just to confirm, this PR doesn't include real implementations of these props. But just demonstrates a way about how to use such props to customize style.

This PR is quite old, and it has unclear status. Do you plan to work on it further or can it be closed?

**Pull request closed without merging**